website_sale_show_stock_preview/controllers/main.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.sale.controllers.variant import VariantController
+
+
+class WebsiteSaleVariantController(VariantController):
+    @http.route(['/sale/get_combination_info_stock_preview'], type='json', auth="public", methods=['POST'], website=True)
+    def get_combination_info_stock_preview(self, product_template_ids, **kw):
```

Comment:
I think that this can be used to get products info that are not set as 'published on the website'...
But this information doesn't identify anything... so I really don't know if it's necessary checking if the product is published or not :/